lib/jsdom/living/streams/ByteLengthQueuingStrategy-impl.js

```diff
@@ -0,0 +1,27 @@
+"use strict";
+
+const sizeFunctionWeakMap = new WeakMap();
+
+exports.implementation = class ByteLengthQueuingStrategyImpl {
+  constructor(globalObject, [{ highWaterMark }]) {
+    this._globalObject = globalObject;
+    this.highWaterMark = highWaterMark;
+  }
+
+  get size() {
+    initializeSizeFunction(this._globalObject);
+    return sizeFunctionWeakMap.get(this._globalObject);
+  }
+};
+
+function initializeSizeFunction(globalObject) {
+  if (sizeFunctionWeakMap.has(globalObject)) {
+    return;
+  }
+
+  // We need to set the 'name' property:
+  // eslint-disable-next-line prefer-arrow-callback
```

Comment:
FWIW you can do

```js
const size = chunk => chunk.byteLength;
sizeFunctionWeakMap.set(globalObject, size);
```

The `size` function's name property will be automatically set.

----

Also, do we need to create the function in `globalObject`, to ensure the prototype is correct?

Reply:
This is straight from the reference impl. Personally I have a mild preference for `function size() {}` over `const size = `, I think it makes what's going on a little more clear.

I think you're right about the prototype, but that's a jsdom-wide problem, isn't it? E.g. `Object.getPrototypeOf(window.Headers) !== window.Function.prototype`. Not sure it's worth hacking around that issue here specifically, unless there's a really simple fix.

Reply:
I think using `function size() { }` is important because per spec the result has a `.prototype` property, whereas arrow functions do not. (I wonder if that's tested.)

Reply:
@ninevra 

> I think you're right about the prototype, but that's a jsdom-wide problem, isn't it?

Ah yes, indeed.

----

@domenic Printing

```js
new ByteLengthQueuingStrategy({ highWaterMark: 100 }).size.prototype
```

gives undefined in Chrome and Firefox. The Streams spec calls CreateBuiltinFunction without calling MakeConstructor, which is the abstract op that adds the `prototype` property. So it would appear that we should in fact use arrow functions after all

Reply:
[Patch submitted upstream](https://github.com/whatwg/streams/pull/1142)